plugins/experimental/carp/CarpConfig.cc

```diff
@@ -0,0 +1,581 @@
+        if(sValue.compare(KEY_MODE_PREREMAP_STR) == 0) {
+          _mode = PRE;
+        } else if(sValue.compare(KEY_MODE_POSTREMAP_STR) == 0) {
+          _mode = POST;
+        } else {
+          TSError("carp: invalid mode in '%s'", buffer);
+        }
+      } else if (sKey.compare(KEY_ALLOWFWDPORT_STR) == 0) {
+        int iPort;
+        if (!getInt(&pTemp, &iPort)) {
+           TSError("carp: error parsing number from '%s'", buffer);
+        }
+        else {
+          TSDebug(DEBUG_TAG_INIT, "Allow forwarding port=%d", iPort);
+           // store the parsed data
+          _allowForwardPort = iPort;
+        }
+      } else if (sKey.compare(KEY_REPLICATIONFACTOR_STR) == 0) {
+        int iFactor;
+        if (!getInt(&pTemp, &iFactor)) {
+           TSError("carp: error parsing number from '%s'", buffer);
+        }
+        else {
+          TSDebug(DEBUG_TAG_INIT, "Replication factor=%d", iFactor);
+           // store the parsed data
+          _replicationFactor = iFactor;
+        }
+ 
+      } else
+        TSError("carp found bad setting under Values section '%s'", buffer);
+      break;
+    }
+    default:
+      // ignore bad cfg_section
+      TSDebug(DEBUG_TAG_INIT, "hit default in switch, ignoring extra input '%s'",buffer);
+      break;
+    };
+  } //while
+
+  if (_healthCheckTimeout > _healthCheckFreq - 1 ) {
+    _healthCheckTimeout = _healthCheckFreq - 1;
+  }
+
+  TSfclose(file);
+
+  if (_blackList.size() && _whiteList.size() ) {
+    TSError("Carp configured with both blacklist and whitelist, blacklist will be ignored");
+  }
+
+  if (_nGroups > _replicationFactor) {
+    TSError("Too many groups configured! Failing config.");
+    return false;
+  } else {
+    TSDebug(DEBUG_TAG_INIT, "Group Config is as follows:");
+    for (map<int, int>::const_iterator it = group_counts.begin(); it != group_counts.end(); it++) {
+       TSDebug(DEBUG_TAG_INIT, "Group %d has %d members.", it->first, it->second);
+       _group_count_list[it->first] = it->second;
+    }
+  }
+
+  return true;
+}
+
+/**********************************************************/
+void
+CarpConfig::addHost(CarpHost* host) {
+  _servers.push_back(host);
+}
+
+void
+CarpConfig::addHealthCheckClient(HttpFetch *client) {
+  client->setHealthcheckTimeout(_healthCheckTimeout);
+  _httpClients.push_back(client);
+
+}
+
+void
+CarpConfig::setPath(string path) {
+  _configPath = path;
+}
+
+string
+CarpConfig::getPath() {
+  return _configPath;
+}
+
+/**********************************************************/
+void *
+CarpConfigHealthCheckThreadStart(void* data)
+{
+  CarpConfigAndHash* cch = static_cast<CarpConfigAndHash *> (data);
+  TSAssert(cch);
+  return cch->_config->run(cch->_hashAlgo);
+}
+
+/**********************************************************/
+bool 
+CarpConfig::isBlackListed(const string& sHost)
+{
+  if(!_blackList.size()) return false;
+  return (_blackList.find(sHost) !=  _blackList.end());
```

Comment:
Not clear it's worth checking for an empty list, as `find` will do that as well.